.github/workflows/testing.yml

```diff
@@ -23,35 +23,6 @@ jobs:
           commit-filter: '[skip ci];[ci skip];[skip github]'
           commit-filter-separator: ';'
 
-  min_requirements:
-    needs: check_skip
-    if: ${{ needs.check_skip.outputs.skip == 'false' }}
-    runs-on: "ubuntu-latest"
-    name: Python 3.8 with minimal requirements
-    defaults:
-      run:
-        shell: bash
-    env:
-        MIN_REQUIREMENTS: true
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
-        with:
-          python-version: 3.8
-        name: 'Setup python'
-      - shell: bash -l {0}
-        run: ./build_tools/github/dependencies.sh
-        name: 'Install dependencies'
-      - shell: bash -l {0}
-        run: ./build_tools/github/install.sh
-        name: 'Install dirty-cat'
-      - shell: bash -l {0}
-        run: ./build_tools/github/test.sh
-        name: 'Run tests'
-      - uses: codecov/codecov-action@v1
-        if: success()
-        name: 'Upload coverage to CodeCov'
-
```

Comment:
By removing this, I believe we don't test the package with the minimal requirements anymore (I don't see a corresponding action in the CI of this PR).

Reply:
Yes we do, the minimal requirements are now the default (see `setup.cfg`).

Reply:
OK, but do we test more recent versions of the dependencies?

Our CI scripts are a bit awkward and do not enable to easily test different combinaison of requirements (and this should chance, see #384 )

Reply:
From what I can tell inspecting the tests of this PR, is seems that every test is now run with the min dependencies.

The tests should really sample a variety of packages versions. Granted, this is awkward to do with our current test system. But I really frown at the idea of merging something that decreases our already-thin test entropy.

The right thing to do would be to see how the version entropy is done in the tests for scikit-learn, and port it here.